sdk/tables/Azure.Data.Tables/src/Extensions/TableEntityExtensions.cs

```diff
@@ -39,6 +39,11 @@ internal static class TableEntityExtensions
                     case byte[] _:
                         annotatedDictionary[prop.Name.ToOdataTypeString()] = TableConstants.Odata.EdmBinary;
                         break;
+                    case BinaryData _:
```

Comment:
I think that you may be able to capture the type conversion here and avoid the cast on L45.

```csharp
case BinaryData binaryData:
    annotatedDictionary[prop.Name.ToOdataTypeString()] = TableConstants.Odata.EdmBinary;
    // Get the bytes from the BinaryData so that it serializes properly.
    annotatedDictionary[prop.Name] = binaryData.ToArray();
    break;
```

Reply:
Good point. I decided to push a PR to Utf8JsonWriterExtensions instead to avoid the call to ToArray which will also avoid the cast. https://github.com/Azure/autorest.csharp/pull/1110